### PR TITLE
add back the app tests

### LINKED ISSUE #14: add basic functional tests
Resurrect the basic application testing from outrigger and add it here.

### PR DESCRIPTION
**Pull request by rlgomes**
this will add back the app tests that will verify the basic usage of
the juttle-engine application

I also added in eslint for the tests and the basis for coverage checking

Fixes #14

### CODE REVIEW

**Comment by rlgomes** [comment]
@juttle/developers app tests for juttle-engine

File: gulpfile.js [hunk 1]
```diff
@@ -1,7 +1,19 @@
 var eslint = require('gulp-eslint');
 var gulp = require('gulp');
+var isparta = require('isparta');
+var istanbul = require('gulp-istanbul');
+var mocha = require('gulp-mocha');
 
-gulp.task('lint', function() {
+gulp.task('lint-test', function() {
```
Comment (demmer) [inline gulpfile.js]: Is there really much benefit to splitting the lint into test and lib?
Seems like we'd do fine with a single task.

Comment (rlgomes) [inline gulpfile.js]: fine by me.

File: package.json [hunk 2]
```diff
@@ -27,14 +27,23 @@
     "daemon": "^1.1.0",
     "express": "^4.13.4",
     "juttle": "^0.4.0",
-    "juttle-service": "^0.1.0",
+    "juttle-service": "juttle/juttle-service",
     "juttle-viewer": "0.0.1",
```
Comment (demmer) [inline package.json]: Maybe change this to juttle/juttle-viewer for now as well so we can pull in fixes. We'll lock down to specific versions before release.

**Comment by demmer** [comment]
@rlgomes good idea.

This looks good once we get a juttle-viewer build that properly handles errors/warnings. Blocked on @mnibecker for that one.

Then you can unskip the tests, verify it works, and close #14.

**Comment by rlgomes** [comment]
@mnibecker

**Comment by demmer** [comment]
@rlgomes According to @mnibecker this should be fixed with juttle-viewer 0.1.0. If we rebase this on top of master it should pick up the correct juttle-viewer distribution. I'll try that now.

**Comment by rlgomes** [comment]
@demmer no worries I can rebase and try it out on my side and then push without the skips if this looks right

**Comment by demmer** [comment]
I'm working on it now... should be ready soon.

**Comment by rlgomes** [comment]
ok just tag me

**Comment by demmer** [comment]
@rlgomes All set... This passes for me locally.

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+host ip address is with:
+
+```
+docker network inspect bridge | grep Gateway
+```
+
+That `Gateway` is your host ip that should be used with the following command:
+
+```
+OUTRIGGER_HOST=172.17.42.1 SELENIUM_REMOTE_URL='http://localhost:4444/wd/hub' gulp test --app
```
Comment (rlgomes) [inline test/README.md]: need to clean up the OUTRIGGER references in here

**Comment by rlgomes** [comment]
+1 LTGM! (just updated the README but no need to wait for the tests to pass)

**Pull request merged**